tests/test_encoder.js

```diff
@@ -0,0 +1,154 @@
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { test } = require('uvu');
+const assert = require('uvu/assert');
+
+const { NbsDecoder, NbsEncoder } = require('..');
+
+const pingType = Buffer.from('8ce1582fa0eadc84', 'hex'); // nuclear hash of 'message.Ping'
+const pongType = Buffer.from('37c56336526573bb', 'hex'); // nuclear hash of 'message.Pong'
+
+function usingTempDir(callback) {
+  const tempDir = fs.mkdtempSync(`${os.tmpdir()}${path.sep}`);
+  try {
+    callback(tempDir);
+  } finally {
+    fs.rmSync(tempDir, { recursive: true });
+  }
+}
+
+test('NbsEncoder constructor throws for invalid arguments', () => {
+  assert.throws(
+    () => {
+      new NbsEncoder();
+    },
+    /missing argument `path`: provide a path to write to/,
+    'NbsEncoder() constructor throws without path argument'
+  );
+
+  assert.throws(
+    () => {
+      new NbsEncoder(1);
+    },
+    /invalid argument `path`: expected string/,
+    'NbsEncoder() constructor throws with incorrect type for path argument'
+  );
+});
+
+test('NbsEncoder write throws for invalid arguments', () => {
+  usingTempDir((dir) => {
+    const file = path.join(dir, 'output.nbs');
+    const encoder = new NbsEncoder(file);
+
+    assert.throws(
+      () => {
+        encoder.write();
+      },
+      /missing argument `packet`: provide a packet to write to the file/,
+      'NbsEncoder write() throws without packet argument'
```

Comment:
Same with the others below

```suggestion
      'NbsEncoder.write() throws without packet argument'
```